.travis.yml

```diff
@@ -18,13 +19,19 @@ jobs:
         - SHA_PR=`git log --oneline -1 | sed -rn "s/.*Merge (.*) into.*/\1/p"`
         - SHA_MAIN=`git log --oneline -1 | sed -rn "s/.*Merge .* into (.*)/\1/p"`
         - scripts/checkpatch.pl --strict --codespell -g $SHA_MAIN..$SHA_PR
-    - name: "sparse check"
+    - name: "Code Scan"
       script:
+        - sudo apt-get install automake ocaml-native-compilers ocaml-findlib
+        - wget https://github.com/coccinelle/coccinelle/archive/1.0.7.tar.gz
+        - tar -xf 1.0.7.tar.gz
+        - cd coccinelle-1.0.7 && ./autogen && ./configure && make && sudo make install
+        - cd -
         - export ARCH=x86_64
         - make defconfig
         - scripts/kconfig/merge_config.sh .config kconfig/base-defconfig kconfig/sof-defconfig kconfig/hdaudio-codecs-defconfig
         - make modules_prepare
         - make SUBDIRS=sound/soc/sof C=2
+        - make coccicheck MODE=report M=sound/soc/sof
```

Comment:
While you are at it you should fix the sparse command and replace SUBDIRS= by M=

Reply:
@plbossart 
Thanks, good point. The log is suggesting to do that.